File: code-ninja-course/CSHARP/Tests/Factorials/FactorialTests.cs

using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.IO;
using System.Text;

namespace LegacyCodeToTestableCode
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class FactorialTests
    {
        [TestMethod]
        public void Factorials()
        {
		    Assert.AreEqual(1, Factorial.calculate(1));
		    Assert.AreEqual(2, Factorial.calculate(2));
        	Assert.AreEqual(6, Factorial.calculate(3));
        	Assert.AreEqual(3628800, Factorial.calculate(10));
        }

        [Ignore]       
        [TestMethod]
        public void FactorialApprovalTests()
        {
            StringBuilder fakeoutput = new StringBuilder();
            Console.SetOut(new StringWriter(fakeoutput));
            Console.SetIn(new StringReader("a\n"));

            Console.WriteLine("1 -> " + Factorial.calculate(1));
            Console.WriteLine("2 -> " + Factorial.calculate(2));
            Console.WriteLine("3 -> " + Factorial.calculate(3));
            Console.WriteLine("10 -> " + Factorial.calculate(10));
            string output = fakeoutput.ToString();
            Approvals.Verify(output);
        }
    }
}

File: code-ninja-course/JAVA/src/test/java/com/codeninja/factorials/FactorialTests.java
package com.codeninja.factorials;

import static org.junit.Assert.*;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.JunitReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Ignore;
import org.junit.Test;

//@UseReporter({DiffReporter.class})
@UseReporter({JunitReporter.class})
public class FactorialTests {

	@Test
	public void factorials() {
		assertEquals(1, Factorial.calculate(1));
		assertEquals(2, Factorial.calculate(2));
		assertEquals(6, Factorial.calculate(3));
		assertEquals(3628800, Factorial.calculate(10));
	}

    @Ignore       
	@Test
	public void factorialApprovalTests() {
		
        StringBuilder sb = new StringBuilder();
        sb.append("1 -> " + Factorial.calculate(1) + "\n");
        sb.append("2 -> " + Factorial.calculate(2)+ "\n");
        sb.append("3 -> " + Factorial.calculate(3)+ "\n");
        sb.append("10 -> " + Factorial.calculate(10)+ "\n");
        
        Approvals.verify(sb.toString());
    }
}     
